### PR TITLE
Implement classical inheritance for Adapters.

### PR DESCRIPTION
**Pull request by jfexyz**
Don't merge yet...

### CODE REVIEW

File: src/Express.js [hunk 1]
```diff
@@ -1,75 +1,82 @@
 'use strict';
 
+import {Router} from 'ferry';
```
Comment (desmondmorris) [inline src/Express.js]: woo hoo!

File: src/Express.js [hunk 2]
```diff
@@ -1,75 +1,82 @@
...
-  route(type, model) {
+  route(action, resource) {
 
-    switch (type) {
+    switch (action) {
 
       case 'index':
         return (req, res)=> {
-          res.send('Index');
+          res.send(action);
```
Comment (desmondmorris) [inline src/Express.js:21]: So I think from the work [here](https://github.com/ferryjs/ferry/pull/17), we will be able to pull the models out of the Ferry object that gets injected?  How will we write tests for this?

**Comment by joshuajabbour** [comment]
Well, we wouldn't pull the models out, we would just use Ferry.storage directly. The storage instance itself would have handlers for each action required (the router has no knowledge of the models).

What should actually happen however, is that ferry and/or the storage engine should inject middleware to handle the action. That's what I'm hoping we can get working cleanly...

File: src/Express.js (hunk 2 above)
Comment (jfexyz) [inline src/Express.js:21]: Well, we wouldn't pull the models out, we would just use Ferry.storage directly. The storage instance itself would have handlers for each action required (the router has no knowledge of the models).

What should actually happen however, is that ferry and/or the storage engine should inject middleware to handle the action. That's what I'm hoping we can get working cleanly...

File: src/Express.js (hunk 1 above)
Comment (jfexyz) [inline src/Express.js]: Yes, it took forever to figure out the import syntax. [Much better info here](http://babeljs.io/docs/usage/modules).

**Pull request merged**